### ISSUE TITLE
Default values for twitterCardType, openGraphType and ChangePropertyValue are not set on node:reapir

### ISSUE THREAD

**Issue by davidspiola** [neos/neos-seo#49]
If the property values twitterCardType, openGraphType and ChangePropertyValue are already set with “empty string”, the new default value “~” will not be set on node:repair. A first attempt to fix this issue I created a node migration.

```
up:
  comments: 'This is the migration to adjust neos seo properties (twitterCardType, openGraphType, xmlSitemapChangeFrequency) accordingly if they are pre set with "empty string"'
  migration:
    -
      filters:
        -
          type: 'NodeType'
          settings:
            nodeType: 'Neos.Neos:Document'
            withSubTypes: true
      transformations:
        -
          type: 'ChangePropertyValue'
          settings:
            property: 'twitterCardType'
            newValue: '~'
        -
          type: 'ChangePropertyValue'
          settings:
            property: 'openGraphType'
            newValue: '~'
        -
          type: 'ChangePropertyValue'
          settings:
            property: 'xmlSitemapChangeFrequency'
            newValue: '~'
down:
  comments: 'No down migration available’
```

If the values are not set correctly editors will see the “You still have changes…” prompt. Saving the change will set the default value accordingly and the issues disappears for the specific node.

![screenshot 2018-01-29 13 44 33](https://user-images.githubusercontent.com/7107011/35553308-a7fca822-0597-11e8-8a85-c84291d791a2.png)

**Comment by kdambekalns** [comment]
@daniellienert I think this is caused by the quoted null keys (`'~'`) from your PR, could you check that?

**Comment by daniellienert** [comment]
😟 grrml will have a look

**Comment by kdambekalns** [comment]
Ok, I found out the following:

- with `~` being used as key for `editorOptions.values`, what you get is `''` when calling `configuration:show`
- with `'~'` being used as key for `editorOptions.values`, what you get is `'~'`  when calling `configuration:show`

Now, when going to any node that has one of the affected options set to `''` this will no longer match. In fact, when saving a node now, the value stored will literally be the string `~`

Neither matches a `defaultValue` of `null`, on a new document a property with `null` as a default value will not be set and immediately trigger changes when being opened in the UI as well. That should be an issue elsewhere, too…

**Comment by kdambekalns** [comment]
Reverting the quoted keys change solves the symptoms of the issue.

**Comment by daniellienert** [comment]
@davidspiola: the symptom is fixed in version 2.0.5
